src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,121 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
+  linear->Reset();
+  RandomInitialization().Initialize(linear->Weight(), outSize, inSize);
+  linear->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Weight().begin(),
+      linear->Weight().end(), linear->Parameters().begin(),
+      linear->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Bias().begin(),
+      linear->Bias().end(), linear->Parameters().begin() + inSize * outSize,
+      linear->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_cols, inSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Parameters().n_rows, inSize * outSize + outSize);
+}
+
+/**
+ * Atrous Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(AtrousConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 1, outSize = 1;
+  size_t kernelWidth = 3, kernelHeight = 3;
```

Comment:
I suggest to use different values with the GCD = 1.

Reply:
Sure. I used these arguments from the actual test of these layers.